uint128_t.include

```diff
@@ -219,12 +219,12 @@ class uint128_t{
 
         template <typename T, typename = typename std::enable_if<std::is_integral<T>::value, T>::type >
         bool operator&&(const T & rhs) const{
-            return static_cast <bool> (*this && rhs);
+            return ((bool) *this && rhs);
         }
 
         template <typename T, typename = typename std::enable_if<std::is_integral<T>::value, T>::type >
         bool operator||(const T & rhs) const{
-            return static_cast <bool> (*this || rhs);
+            return ((bool) *this || rhs);
```

Comment:
the `bool` cast applies to `*this`, but you really want it to apply to the whole expression.
what's the rationale for this change?

Reply:
in an ideal world, this fix would come with a unit test exercising the bug

Reply:
Since `uint256_t` has the same code, I figured this was simply a forgotten fix.

Reply:
The `||` operator already returns a `bool`, so I don't understand why you need a `static_cast<bool>` there in the first place..

Reply:
If you ask me, the overriding of `operator||` and `operator&&` in `uint128_t` and `uint256_t` is unnecessary, `operator bool()` is all you need to make it work.

Reply:
but `rhs` is a template. it's not necessarily a `uint256`, it could also be a `uint64_t`, and its `&&` operator does not return a `bool`. (in fact, it's a bit strange to have these operators return a `bool`, it is not what I would expect from bit-wise operators.

Reply:
Please use `static_cast` instead of C style casting.

Reply:
How about removing the `||` and `&&` operators altogether? They don't implement any specific behavior.

Reply:
I can make a new PR with just the `_UINT128_T_STATIC` change.